### PR TITLE
feat(pie-storybook): DWS-000 added documentation page for button vari…

### PR DESCRIPTION
**Pull request by LTurns**
## Describe your changes (can list changeset entries if preferable)

[Added] - Documentation page for button variants with code snippets

In order to embed code snippets of the different variants in our pie design system, we need to create a docs page. Storybook does provide autodocs out of the box, but a bespoke page gave more flexibility in terms of adding content and separating the variant code snippets from the stories.

Tested in local branch with code button template:

https://github.com/justeattakeaway/pie/assets/49618712/644a09ee-6061-42df-b6b7-d8d62202b091

https://pr982-storybook.pie.design/?path=/docs/button--variants

## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

**Review by JoshuaNg2332 — APPROVED** [review]

**Review by maledr5 — DISMISSED** [review]
Awesome work! 🤩

**Comment by ashleynolan** [comment]
This looks awesome 👏 👏 👏 

Only tiny thing I could notice – When viewing the link in the PR, I see this spacing on the component storybook. Looks like that's not shown in your demo though on the docs site, but wondering if there's some extra whitespace somewhere?

Screenshot 2023-11-08 at 17 23 52

File: apps/pie-storybook/stories/pie-button.stories.tsx [hunk 1]
```diff
@@ -170,24 +170,24 @@ const Template: TemplateFunction<ButtonProps> = ({
...
+    ?isFullWidth="${isFullWidth}"
+    name=${name || nothing}
+    value=${value || nothing}
+    formaction=${formaction || nothing}
+    formenctype=${formenctype || nothing}
+    formmethod=${formmethod || nothing}
+    formtarget=${formtarget || nothing}
+    ?formnovalidate="${formnovalidate}">
+    ${iconPlacement ? html`<icon-plus-circle slot="icon"></icon-plus-circle>` : nothing}
+    ${slot}
```
Comment (ashleynolan) [inline apps/pie-storybook/stories/pie-button.stories.tsx:189]: Just to add – I think it's to do with this section. If there's no icon, you'll get an extra line of whitespace in the docs. Can probably fix this by doing:
```suggestion
    ?formnovalidate="${formnovalidate}">${
    iconPlacement ? html`<icon-plus-circle slot="icon"></icon-plus-circle>` : nothing}
    ${slot}
```

**Comment by LTurns** [comment]
> This looks awesome 👏 👏 👏
> 
> Only tiny thing I could notice – When viewing the link in the PR, I see this spacing on the component storybook. Looks like that's not shown in your demo though on the docs site, but wondering if there's some extra whitespace somewhere?
> 
> Screenshot 2023-11-08 at 17 23 52

ah thank you! I know why, I just need to change the indentation of the slot. Made the change locally when testing but must have missed it on the PR 🤦 I'll make that change in my next follow up PR which should be ready quickly after this one 😄

File: apps/pie-storybook/stories/pie-button.stories.tsx (hunk 1 above)
Comment (LTurns) [reply apps/pie-storybook/stories/pie-button.stories.tsx]: all fixed 😄 thank you! https://pr982-storybook.pie.design/?path=/docs/button--variants

**Review by FayeCarter — APPROVED** [review]

**Review by JoshuaNg2332 — APPROVED** [review]

**Review by maledr5 — APPROVED** [review]

**Pull request merged**